src/extensions/replay/sessionrecording.ts

```diff
@@ -909,11 +955,19 @@ export class SessionRecording {
             this._pageViewFallBack()
         }
 
+        // Check if the URL matches any trigger patterns
+        this._checkUrlTrigger()
+
         // we're processing a full snapshot, so we should reset the timer
         if (rawEvent.type === EventType.FullSnapshot) {
             this._scheduleFullSnapshot()
         }
 
+        // Clear the buffer if waiting for a trigger, and only keep data from after the current full snapshot
+        if (rawEvent.type === EventType.FullSnapshot && this.urlTriggerStatus === 'pending') {
+            this.clearBuffer()
```

Comment:
i guess this is still to-be-implemented behaviour?

(adding note just in case but assuming so since still draft)

Reply:
Hmm, I was assuming this already does this, since the flow is essentiually: 

- We receive a new snapshot
- We clear the buffer
- We add the full snapshot to the buffer

Am I missing something here?

Reply:
No, I was being silly! You're totally right